linetools/spectra/xspectrum1d.py

```diff
@@ -182,6 +192,15 @@ def __init__(self, wave, flux, sig=None, co=None, units=None, select=0,
             if wave.shape[0] != sig.shape[0]:
                 raise IOError("Shape of wave and sig vectors must be identical")
             self.data['sig'] = np.reshape(sig, (self.nspec, self.npix))
+            if masking != 'None':
+                for kk in range(self.nspec):
+                    sigval = np.where(self.data['sig'][kk] > 0.)[0]
+                    for key in self.data.dtype.names:
+                        if masking == 'edges':
+                            self.data[key][kk][0:sigval[0]].mask = True
+                            self.data[key][kk][sigval[-1]+1:].mask = True
+                        elif masking == 'all':
+                            self.data[key][kk][sigval].mask = True
```

Comment:
This approach will mask out the good pixels and leave in the bad ones. 'True' -> 'False'